### PR TITLE
Fixes #218: Allow rendering description from schema for all fields.

### LINKED ISSUE #218: Description for boolean field
Now no description for boolean type is displayed - only title and rewriting the DescriptionField doesn't affect boolean type.
Is it possible to add support of description to boolean type?

### PR DESCRIPTION
**Pull request by amarnus**
Helps with #218

### CODE REVIEW

File: src/components/fields/SchemaField.js [hunk 1]
```diff
@@ -95,6 +97,7 @@ function Wrapper({
   return (
     <div className={classList}>
       {displayLabel && label ? getLabel(label, required, id) : null}
+      <DescriptionField id={ `${id}__description` } description={ description } />
```
Comment (n1k0) [inline src/components/fields/SchemaField.js]: nit: <code>{`${id}__description`}</code> (no space within curly brackets)

File: src/components/fields/SchemaField.js [hunk 2]
```diff
@@ -156,6 +160,7 @@ function SchemaField(props) {
   return (
     <Wrapper
       label={props.schema.title || schema.title || name}
+      description={props.schema.description}
```
Comment (n1k0) [inline src/components/fields/SchemaField.js]: We should rather retrieve this from the resolved schema, so `schema.description` here.

**Comment by n1k0** [comment]
Would you mind adding a few tests? Thanks!

**Comment by amarnus** [comment]
Done.

File: test/SchemaField_test.js [hunk 3]
```diff
@@ -137,4 +137,46 @@ describe("SchemaField", () => {
         .to.have.length.of(1);
     });
   });
+
+  describe("description support", () => {
+    const schema = {
+      type: "object",
+      properties: {
+        foo: {type: "string",description:"A Foo field"},
```
Comment (n1k0) [inline test/SchemaField_test.js]: nit: missing space after comma

File: test/SchemaField_test.js [hunk 4]
```diff
@@ -137,4 +137,46 @@ describe("SchemaField", () => {
...
+        bar: {type: "string"}
+      }
+    };
+    it("should render description if available from the schema", () => {
+      const {node} = createFormComponent({schema});
+      expect(node.querySelectorAll("div#root_foo__description"))
+          .to.have.length.of(1);
```
Comment (n1k0) [inline test/SchemaField_test.js]: nit: 2 spaces indent here

**Comment by amarnus** [comment]
Thanks for highlighting the styling issues. Resolved now. `npm run lint` didn't catch them though. Will be good to add these rules in there, I feel?

File: test/SchemaField_test.js [hunk 5]
```diff
@@ -137,4 +137,46 @@ describe("SchemaField", () => {
...
+    });
+
+    it("should render description if available from a referenced schema", () => {
+      // Overriding.
+      const schema = {
+        type: "object",
+        properties: {
+          foo: {"$ref": "#/definitions/foo"},
```
Comment (n1k0) [inline test/SchemaField_test.js]: nit: `$ref` (no quotes) works too

File: test/SchemaField_test.js [hunk 6]
```diff
@@ -137,4 +137,46 @@ describe("SchemaField", () => {
...
+          foo: {
+            type: "string",
+            description: "A Foo field"
+          }
+        }
+      };
+      const {node} = createFormComponent({schema});
+      const matches = node.querySelectorAll("div#root_foo__description");
+      expect(matches).to.have.length.of(1);
+      expect(matches[0].innerHTML).to.equal("A Foo field");
```
Comment (n1k0) [inline test/SchemaField_test.js]: How about using the `textContent` property instead of `innerHTML`?

File: src/components/fields/SchemaField.js [hunk 7]
```diff
@@ -156,6 +160,7 @@ function SchemaField(props) {
   return (
     <Wrapper
       label={props.schema.title || schema.title || name}
+      description={props.schema.description || schema.description}
```
Comment (n1k0) [inline src/components/fields/SchemaField.js]: I'd probably favor the resolved schema first, what do you think?

**Comment by n1k0** [comment]
> npm run lint didn't catch them though. Will be good to add these rules in there, I feel?

If you know about what the rules to add are, please add them :)

File: src/components/fields/SchemaField.js (hunk 7 above)
Comment (amarnus) [inline src/components/fields/SchemaField.js]: I don't have a preference, actually. Just kept it consistent with the previous expression that sets the `label` prop.

**Comment by amarnus** [comment]
>If you know about what the rules to add are, please add them :)

Will do. As another pull request.

File: test/SchemaField_test.js [hunk 8]
```diff
@@ -137,4 +137,46 @@ describe("SchemaField", () => {
...
+    };
+    it("should render description if available from the schema", () => {
+      const {node} = createFormComponent({schema});
+      expect(node.querySelectorAll("div#root_foo__description"))
+        .to.have.length.of(1);
+    });
+
+    it("should render description if available from a referenced schema", () => {
+      // Overriding.
+      const schema = {
```
Comment (n1k0) [inline test/SchemaField_test.js]: I'm surprised you can reassign a `const` here. You should probably name this specific schema differently instead.

**Comment by n1k0** [comment]
Side note, please avoid squashing and keep your commits atomic on the PR so I can review them individually. Squashing is enabled for this repo so a single commit will eventually land anyway :)

**Comment by amarnus** [comment]
Noted. It's just that I didn't want the intermediate commits on my fork either. But seeing that you have left comments on them, I guess I shouldn't be squashing them. Got it.

**Comment by amarnus** [comment]
When you have a spare moment later, can restart the Travis build? It failed because of issues with `npm`.

**Comment by n1k0** [comment]
Jobs restarted.

File: test/SchemaField_test.js [hunk 9]
```diff
@@ -137,4 +137,46 @@ describe("SchemaField", () => {
...
+
+  describe("description support", () => {
+    const schema = {
+      type: "object",
+      properties: {
+        foo: {type: "string", description:"A Foo field"},
+        bar: {type: "string"}
+      }
+    };
+    it("should render description if available from the schema", () => {
```
Comment (n1k0) [inline test/SchemaField_test.js]: nit: insert blank line here

**Comment by n1k0** [comment]
Strange issue with travis it seems, where fetching this url https://registry.npmjs.org/babel-plugin-transform-object-rest-spread/-/babel-plugin-transform-object-rest-spread-6.8.0.tgz fails with a `502` while

```
$ curl -I https://registry.npmjs.org/babel-plugin-transform-object-rest-spread/-/babel-plugin-transform-object-rest-spread-6.8.0.tgz
HTTP/1.1 200 OK
```

**Comment by n1k0** [comment]
Perfect, I'm landing this. Again, thanks for contributing!

**Comment by n1k0** [comment]
Released in [v0.36.0](https://github.com/mozilla-services/react-jsonschema-form/releases/tag/v0.36.0).

**Pull request merged**